dove/tests/test_cmd_dove_init.rs

```diff
@@ -0,0 +1,347 @@
+};
+
+/// $ dove init
+#[test]
+fn test_cmd_dove_init_without_arguments() {
+    // Project name and path
+    let project_name = "demoproject_36";
+    let project_folder = project_start_for_init(project_name);
+
+    execute_dove_at(&project_folder, &["dove", "init"]);
+    // Check config
+    let package = read_to_string(project_folder.join("Dove.toml"))
+        .unwrap()
+        .parse::<Value>()
+        .unwrap()
+        .get("package")
+        .unwrap()
+        .clone();
+
+    assert!(
+        package
+            .get("name")
+            .unwrap()
+            .to_string()
+            .contains(project_name),
+        "Dove.toml: invalid name",
+    );
+
+    assert!(
+        package.get("dialect").unwrap().to_string().contains("pont"),
+        "Dove.toml: invalid dialect",
+    );
+
+    set_dependencies_local_move_stdlib(&project_folder);
+    project_build(&project_folder);
+    project_remove(&project_folder);
+}
+/// $ dove init -d ###
+#[test]
+fn test_cmd_dove_init_with_dialect() {
+    // Project name and path
+    let project_name = "demoproject_43";
+    let project_folder = project_start_for_init(project_name);
+
+    for dialect in &["pont", "diem", "dfinance"] {
+        execute_dove_at(&project_folder, &["dove", "init", "-d", dialect]);
+        // Check config
+        let package = read_to_string(project_folder.join("Dove.toml"))
+            .unwrap()
+            .parse::<Value>()
+            .unwrap()
+            .get("package")
+            .unwrap()
+            .clone();
+
+        assert!(
+            package
+                .get("name")
+                .unwrap()
+                .to_string()
+                .contains(project_name),
+            "Dove.toml: invalid name",
+        );
+
+        assert!(
+            package
+                .get("dialect")
+                .unwrap()
+                .to_string()
+                .contains(dialect),
+            "Dove.toml: invalid dialect",
+        );
+
+        set_dependencies_local_move_stdlib(&project_folder);
+        project_build(&project_folder);
+        project_remove(&project_folder);
+    }
+}
+/// $ dove init -d dfinance -a ###
+#[test]
+fn test_cmd_dove_init_dfinance_with_address() {
+    // Project name and path
+    let project_name = "demoproject_44";
+    let (_, project_folder) = project_start(project_name);
+    project_remove(&project_folder);
+
+    for address in &["0x1", "wallet1me0cdn52672y7feddy7tgcj6j4dkzq2su745vh"] {
+        // Create project directory
+        create_dir_all(&project_folder).unwrap_or_else(|_| {
+            panic!(
+                "Failed to create directory: {}",
+                project_folder.to_str().unwrap_or(" - "),
+            )
+        });
+        execute_dove_at(
+            &project_folder,
+            &["dove", "init", "-d", "dfinance", "-a", address],
+        );
+
+        // Check config
+        let package = read_to_string(project_folder.join("Dove.toml"))
```

Comment:
Extract reading+parsing of `Dove.toml` file into its own utility `read_dove_toml(PATH)` ?